Undefine altivec.h bool and vector, otherwise these conflict with the c++11 standard defintions
and that causes the build to fail on ppc64le.
Patch by Simon Branford, University of Birmingham
diff -aur pocl-1.4.orig/include/CL/cl_platform.h pocl-1.4/include/CL/cl_platform.h
--- pocl-1.4.orig/include/CL/cl_platform.h	2020-01-03 09:55:56.986465000 +0000
+++ pocl-1.4/include/CL/cl_platform.h	2020-01-03 09:59:18.058173021 +0000
@@ -356,6 +356,8 @@
 /* Define basic vector types */
 #if defined( __VEC__ )
    #include <altivec.h>   /* may be omitted depending on compiler. AltiVec spec provides no way to detect whether the header is required. */
+   #undef bool
+   #undef vector
    typedef __vector unsigned char     __cl_uchar16;
    typedef __vector signed char       __cl_char16;
    typedef __vector unsigned short    __cl_ushort8;
